azutils/azutils/MountClient.py

```diff
@@ -62,25 +62,49 @@ def read(cls, source_path: str, source_format: str, options:dict = None):
         AKVClient.dbutils.fs.ls(directory)
         #check if file exists
         if source_file in [file.name for file in AKVClient.dbutils.fs.ls(directory)]:
-          if options == None:
-            if source_format == "csv":
-              #reads csv file with header if file type is csv and reader options are not given
-              options = {"header": True}
+          if schema == None:
+            if options == None:
+              if source_format == "csv":
```

Comment:
This has started to create very deeply nested IF statements that also re-peat a lot of code. I'm thinking if you restructured this function to work with a single output (only `return` once) you should be able to clean this up significantly while making it highly readable.